File: LogicadeProgramacao/Estrutura de Decisão/2. Java/CupomFiscal.java
import java.util.Scanner;

public class CupomFiscal {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        //Tabela de preços
        double fileDuploAte5Kg = 4.90;
        double fileDuploMaisDe5Kg = 5.80;
        double alcatraAte5Kg = 5.90;
        double alcatraMaisDe5Kg = 6.80;
        double picanhaAte5Kg = 6.90;
        double picanhaMaisDe5Kg = 7.80;
        
        System.out.println("Bem-vindo ao Hipermercado Tabajara!");
        System.out.println("Por favor, selecione o tipo de carne desejado:");
        System.out.println("1 - File Duplo");
        System.out.println("2 - Alcatra");
        System.out.println("3 - Picanha");
        int tipoCarne = sc.nextInt();
        
        System.out.println("Quantos quilos de carne deseja comprar?");
        double quantidade = sc.nextDouble();
        
        double precoTotal = 0;
        String tipoPagamento = "";
        double valorDesconto = 0;
        double valorAPagar = 0;
        
        //Calcula o preço total de acordo com o tipo de carne e a quantidade comprada
        switch (tipoCarne) {
            case 1:
                if (quantidade <= 5) {
                    precoTotal = quantidade * fileDuploAte5Kg;
                } else {
                    precoTotal = quantidade * fileDuploMaisDe5Kg;
                }
                tipoPagamento = "à vista";
                break;
            case 2:
                if (quantidade <= 5) {
                    precoTotal = quantidade * alcatraAte5Kg;
                } else {
                    precoTotal = quantidade * alcatraMaisDe5Kg;
                }
                tipoPagamento = "à vista";
                break;
            case 3:
                if (quantidade <= 5) {
                    precoTotal = quantidade * picanhaAte5Kg;
                } else {
                    precoTotal = quantidade * picanhaMaisDe5Kg;
                }
                tipoPagamento = "à vista";
                break;
            default:
                System.out.println("Opção inválida!");
                System.exit(0);
        }
        
        System.out.println("Deseja pagar com cartão Tabajara? (s/n)");
        String opcaoCartao = sc.next();
        if (opcaoCartao.equalsIgnoreCase("s")) {
            tipoPagamento = "com cartão Tabajara";
            valorDesconto = precoTotal * 0.05;
        }
        
        valorAPagar = precoTotal - valorDesconto;
        
        System.out.println("Cupom Fiscal");
        System.out.println("Tipo de carne: " + tipoCarne);
        System.out.println("Quantidade: " + quantidade + " Kg");
        System.out.println("Preço total: R$" + precoTotal);
        System.out.println("Tipo de pagamento: " + tipoPagamento);
        System.out.println("Valor do desconto: R$" + valorDesconto);
        System.out.println("Valor a pagar: R$" + valorAPagar);
        
        sc.close();
    }

}

File: LogicadeProgramacao/Estrutura de Decisão/3. Python/CupomFiscal.py
print("=== Hipermercado Tabajara ===")
print("Tipos de carne em promoção:")
print("1 - File Duplo")
print("2 - Alcatra")
print("3 - Picanha")

tipo_carne = int(input("Digite o número do tipo de carne desejado: "))
quantidade = float(input("Digite a quantidade desejada (em kg): "))

if tipo_carne == 1:
    preco_kg = 4.90
    tipo = "File Duplo"
elif tipo_carne == 2:
    preco_kg = 5.90
    tipo = "Alcatra"
elif tipo_carne == 3:
    preco_kg = 6.90
    tipo = "Picanha"
else:
    print("Opção inválida, programa encerrado.")
    exit()

if quantidade <= 5:
    preco_total = preco_kg * quantidade
else:
    preco_total = 5 * preco_kg + (quantidade - 5) * (preco_kg + 0.80) # acima de 5 kg, acrescenta R$ 0,80 por kg extra

forma_pagamento = input("Digite a forma de pagamento (1 - Dinheiro / 2 - Cartão Tabajara): ")
if forma_pagamento == "2":
    desconto = 0.05 * preco_total
else:
    desconto = 0

print("\nCupom fiscal:")
print("Tipo de carne:", tipo)
print("Quantidade:", quantidade, "kg")
print("Preço total: R$ %.2f" % preco_total)
print("Forma de pagamento:", "Dinheiro" if forma_pagamento == "1" else "Cartão Tabajara")
print("Desconto: R$ %.2f" % desconto)
print("Valor a pagar: R$ %.2f" % (preco_total - desconto))
